### PR TITLE
Ps 43 search filters and search page

### PR DESCRIPTION
**Pull request by noFrostoo**
Checkout to page and see how it works, 

Done: 
* search endpoint, 
* best endpoint, 
* popular endpoint, 
* search crud
* search page 
* autocomplete
* reworked routes,
* appbar reacts to logged/logged out
* added snackbar to errors
*

### CODE REVIEW

File: services/backend/app/core/tasks.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+from app.main import app
+from fastapi_utils.tasks import repeat_every
+from app.api import deps
+from app import crud
+from app.models.recipe import Recipe
+
+@app.on_event("startup")
+@repeat_every(seconds=60*60*8) 
+def recalculatePopularity() -> None:
+    db = deps.get_db() #pls work
```
Comment (lorlen) [inline services/backend/app/core/tasks.py]: Depends can be added everywhere IIRC

File: services/backend/app/api/endpoints/recipes.py [hunk 2]
```diff
@@ -1,25 +1,66 @@
...
+    #TODO use enum
+    if sort not in ["popular", "views", "best", "nosort"]:
+        raise HTTPException(
+            status_code=400,
+            detail="Bad tags connect",
+        )
+    
+    keywords = keywords.replace("+", " ")
+
+    query = crud.recpie.start_query(db)
```
Comment (lorlen) [inline services/backend/app/api/endpoints/recipes.py:38]: s/recpie/recipe/

File: services/backend/app/crud/crud_stats.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+from sqlalchemy.orm import Session, Query
+from sqlalchemy.sql.functions import func
+
+from app.crud.base import CRUDBase
+from app.models.stats import Stats
+
+class CRUDStats(CRUDBase[Stats]):
+
+    def get_
```
Comment (lorlen) [inline services/backend/app/crud/crud_stats.py]: Better to comment this and add `pass` after class declaration

Comment (noFrostoo) [reply services/backend/app/crud/crud_stats.py]: done allready

File: services/backend/app/api/endpoints/recipes.py (hunk 2 above)
Comment (noFrostoo) [reply services/backend/app/api/endpoints/recipes.py]: done

File: services/backend/app/core/tasks.py (hunk 1 above)
Comment (noFrostoo) [reply services/backend/app/core/tasks.py]: done

**Review by lorlen — APPROVED** [review]

**Pull request merged**